### PR TITLE
added JWT validation, /login endpoint

### PR DESCRIPTION
**Pull request by maxymmusiienko**

### CODE REVIEW

File: src/main/java/com/example/bookshop/dto/UserLoginRequestDto.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import org.hibernate.validator.constraints.Length;
+
+@Data
+public class UserLoginRequestDto {
+    @NotBlank
+    @Length(max = 100)
+    @Email
+    private String email;
+    @NotBlank
+    @Length(min = 8, max = 25)
```
Comment (Sanderfuck) [inline src/main/java/com/example/bookshop/dto/UserLoginRequestDto.java]: Can we use without @NotBlank annotation?

File: src/main/java/com/example/bookshop/service/AuthenticationServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Service;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Autowired
+    private JwtUtil jwtUtil;
+
```
Comment (Sanderfuck) [inline src/main/java/com/example/bookshop/service/AuthenticationServiceImpl.java]:
```suggestion
```

File: src/main/java/com/example/bookshop/service/AuthenticationServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Service;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Autowired
+    private JwtUtil jwtUtil;
```
Comment (Sanderfuck) [inline src/main/java/com/example/bookshop/service/AuthenticationServiceImpl.java]: Try to inject by constructor and you can use @requiredArgsConstructor of lombok for this but fields should be final

File: src/test/resources/application.properties [hunk 4]
```diff
@@ -3,3 +3,6 @@ spring.datasource.driverClassName=org.h2.Driver
 spring.datasource.username=sa
 spring.datasource.password=password
 spring.jpa.database-platform=org.hibernate.dialect.H2Dialect
+
+jwt.expiration=6000000000000000
```
Comment (Sanderfuck) [inline src/test/resources/application.properties]: In test you don`t need real duration

File: src/main/java/com/example/bookshop/controller/AuthenticationController.java [hunk 5]
```diff
@@ -17,6 +20,15 @@ public class AuthenticationController {
     @Autowired
     private UserService userService;
```
Comment (Sanderfuck) [inline src/main/java/com/example/bookshop/controller/AuthenticationController.java]:
```suggestion
```

**Review by Sanderfuck — CHANGES_REQUESTED** [review]

**Review by olekskov — APPROVED** [review]

**Review by Ivan95kos — APPROVED** [review]
Great job

**Pull request merged**